src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -251,24 +250,52 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   (moduleConfig as any).preloadPlugins = [];
 
   let resourcesLoaded = 0;
-  function setProgress(){
-      resourcesLoaded++;
-      const percentage = resourcesLoaded / totalResources.length * 100;
-      document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
-      document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  function setProgress() {
+    resourcesLoaded++;
+    const percentage = resourcesLoaded / totalResources.length * 100;
+    document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
+    document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  }
+
+  const typesMap: { [key: string]: WebAssemblyBootResourceType | undefined } = {
+    'assembly': 'assembly',
+    'pdb': 'pdb',
+    'icu': 'globalization',
+    'dotnetwasm': 'dotnetwasm',
+  };
+
+  // it would not `loadResource` on types for which there is no typesMap mapping
+  const downloadResource = (request: ResourceRequest): LoadingResource | undefined => {
+    const type = typesMap[request.behavior];
+    if (type !== undefined) {
+      const loaded = resourceLoader.loadResource(request.name, request.resolvedUrl!, request.hash!, type);
```

Comment:
```suggestion
      const loadingResource = resourceLoader.loadResource(request.name, request.resolvedUrl!, request.hash!, type);
```

... to have the same tense as the type name

Reply:
Also, why do we need the `!` after `resolvedUrl` and `hash`? If they are definitely set, can we change the upstream type definition to reflect that? Otherwise it's hard to reason about why this code is valid.

Reply:
The runtime doesn't always have `hash` and `resolvedUrl`. We have two similar typescript types here.